server/broadcast.go

```diff
@@ -604,3 +630,26 @@ func updateSession(sess *BroadcastSession, res *ReceivedTranscodeResult) *Broadc
 
 	return newSess
 }
+
+func refreshSession(sess *BroadcastSession) (*BroadcastSession, error) {
+	uri, err := url.Parse(sess.OrchestratorInfo.Transcoder)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), refreshTimeout)
```

Comment:
Let's add a test for the timeout. A stub value for `refreshTimeout` can be used in the test i.e. a few milliseconds.

Reply:
not sure how to test for a timeout here

Reply:
```golang
oldRefreshTimeout = refreshTimeout
defer func() { refreshTimeout = oldRefreshTimeout }()
refreshTimeout = 100 * time.Millisecond

oldGetOrchestratorInfoRPC = getOrchestratorInfoRPC
defer func() { getOrchestratorInfoRPC = oldGetOrchestratorInfoRPC }()
getOrchestratorInfoRPC = func(ctx context.Context, bcast common.Broadcaster, serv *url.URL) (*net.OrchestratorInfo, error) {
    // Wait until the refreshTimeout has elapsed
    <-ctx.Done()

    return nil, errors.New("context timeout")
}

_ err := refreshSession(..)
// Test for a timeout related error here
```

Reply:
Thanks for the example ! 

Added in 913ef20e